packages/blade/src/components/Tooltip/Tooltip.web.tsx

```diff
@@ -0,0 +1,153 @@
+import { TooltipProps } from './types';
+import { TooltipContent } from './TooltipContent';
+import { ARROW_HEIGHT, ARROW_WIDTH } from './constants';
+import { useTheme } from '~components/BladeProvider';
+import BaseBox from '~components/Box/BaseBox';
+import { makeAccessible, makeBorderSize } from '~utils';
+import { useId } from '~src/hooks/useId';
+import { Box } from '~components/Box';
+
+const TooltipInteractiveWrapper = styled(Box)((props) => {
+  return {
+    display: 'inline-block',
+    '&:focus': {
+      borderRadius: makeBorderSize(props.theme.border.radius.medium),
+      // TODO: Replace with focus outline token
+      outline: `1px solid ${props.theme.colors.surface.background.level1.lowContrast}`,
+      boxShadow: `0px 0px 0px 4px ${props.theme.colors.brand.primary[400]}`,
+    },
+  };
+});
+
+const Tooltip = ({
+  content,
+  children,
+  placement = 'top',
+  shouldWrapChildren,
+  onOpen,
+  onClose,
+}: TooltipProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const id = useId();
+  const [isOpen, setIsOpen] = React.useState(false);
+  const arrowRef = React.useRef<SVGSVGElement>(null);
+
+  const GAP = theme.spacing[2];
+  const [side, alignment] = placement.split('-') as [Side, Alignment];
+  const isHorizontal = side === 'left' || side === 'right';
+  const isCrossAxis = side === 'right' || side === 'bottom';
+  const hasAlignment = alignment === 'start' || alignment === 'end';
+
+  const { refs, floatingStyles, context } = useFloating({
+    placement,
+    open: isOpen,
+    strategy: 'fixed',
+    onOpenChange: (open) => {
+      if (open) {
+        setIsOpen(true);
+        onOpen?.();
+      } else {
+        setIsOpen(false);
+        onClose?.();
+      }
+    },
+    middleware: [
+      shift({ crossAxis: false, padding: GAP }),
+      flip({ padding: GAP }),
+      offset(GAP + ARROW_HEIGHT),
+      arrow({
+        element: arrowRef,
+        padding: isHorizontal ? 0 : ARROW_WIDTH,
+      }),
+    ],
+  });
+
+  const { isMounted, styles } = useTransitionStyles(context, {
+    duration: theme.motion.duration.quick,
+    initial: {
+      opacity: 0,
+      transform: `translate${isHorizontal ? 'X' : 'Y'}(${isCrossAxis ? -10 : 10}px)`,
+    },
+  });
+
+  useDelayGroup(context, { id });
+  const { delay } = useDelayGroupContext();
+  const hover = useHover(context, {
+    delay,
+    move: false,
+  });
+  const focus = useFocus(context);
+  const role = useRole(context, { role: 'tooltip' });
+  const { getReferenceProps, getFloatingProps } = useInteractions([role, hover, focus]);
+
+  return (
+    <>
+      {shouldWrapChildren ? (
+        <TooltipInteractiveWrapper
+          tabIndex={0}
+          ref={refs.setReference as never}
+          {...makeAccessible({ label: content })}
+          {...getReferenceProps()}
+        >
+          {children}
+        </TooltipInteractiveWrapper>
+      ) : (
+        React.cloneElement(children, { ref: refs.setReference, ...getReferenceProps() })
```

Comment:
Can you add a comment on why we need to clone element?

Reply:
Added